Sources/Core/Video/VideoDecoding.swift

```diff
@@ -0,0 +1,85 @@
+// The MIT License (MIT)
+//
+// Copyright (c) 2015-2021 Alexander Grebenyuk (github.com/kean).
+
+
+#if !os(watchOS)
+
+import Foundation
+import AVKit
+
+extension ImageType {
+    public static let mp4: ImageType = "public.mp4"
+}
+
+extension ImageDecoders {
+    public final class Video: ImageDecoding, ImageDecoderRegistering {
+        private var didProducePreview = false
+
+        static func isVideo(_ data: Data) -> Bool {
+            match(data, offset: 4, [0x66, 0x74, 0x79, 0x70])
+        }
+
+        public var isAsynchronous: Bool {
+            false
+        }
+
+        public init?(data: Data, context: ImageDecodingContext) {
+            guard Video.isVideo(data) else { return nil }
+        }
+
+        public init?(partiallyDownloadedData data: Data, context: ImageDecodingContext) {
+            guard Video.isVideo(data) else { return nil }
+        }
+
+        public func decode(_ data: Data) -> ImageContainer? {
+            ImageContainer(image: PlatformImage(), type: .mp4, data: data)
+        }
+
+        public func decodePartiallyDownloadedData(_ data: Data) -> ImageContainer? {
+            guard !didProducePreview else {
+                return nil // We only need one preview
+            }
+            guard let preview = makePreview(for: data) else {
+                return nil
+            }
+            didProducePreview = true
+            return ImageContainer(image: preview, type: .mp4, isPreview: true, data: data)
+        }
+
+        private static var isRegistered: Bool = false
+
+        public static func register() {
```

Comment:
This decoder can be registered by default in `ImageDecoderRegistry` which NukeUI didn't have access to.